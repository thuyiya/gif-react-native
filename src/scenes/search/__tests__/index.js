/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import { VIEW_CONST } from "_constants";
import SearchScreen from "_scenes/search";
import { Provider } from "_contexts";

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  ...props,
});

describe("SCREEN: Search", () => {
  const props = createTestProps({});
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Provider value={{ state: { search: "A" } }}>
        <SearchScreen {...props} />
      </Provider>
    );
  });

  it("enders correctly react-test-renderer", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Child Element Getting Rendered Correctly", () => {
    expect(
      wrapper.find({ testID: VIEW_CONST.SEARCH.flatListTestID }).length
    ).toBe(0);
  });
});
