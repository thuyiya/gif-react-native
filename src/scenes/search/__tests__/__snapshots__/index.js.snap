// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCREEN: Search enders correctly react-test-renderer 1`] = `ShallowWrapper {}`;
